src/types/parser.ts

```diff
@@ -56,17 +57,54 @@ type ParsedBaseEntity = {
 export type ParsedAuthor = ParsedBaseEntity & {
 	beginDate?: string;
 	endDate?: string;
-	type?: 'Person';
+	ended: boolean;
+	type?: string;
+	typeId?: number;
+	genderId?: number;
+	beginAreaId?: number;
+	endAreaId?: number;
 };
 
 export type ParsedEdition = ParsedBaseEntity & {
+	editionGroupBbid: string;
+	width?: number;
+	height?: number;
+	depth?: number;
+	weight?: number;
+	pages?: number;
+	formatId?: number;
+	statusId?: number;
 	languages?: FormLanguageT[];
 	releaseEvents?: FormReleaseEventT[];
 };
 
-export type ParsedWork = ParsedBaseEntity;
+export type ParsedEditionGroup = ParsedBaseEntity & {
+	typeId?: number;
+};
+
+export type ParsedPublisher = ParsedBaseEntity & {
+	typeId?: number;
+	areaId?: number;
+	beginDate?: string;
+	endDate?: string;
+	ended: boolean;
+};
+
+export type ParsedSeries = ParsedBaseEntity & {
+
+	/** Type of the items in the series. */
+	entityType: EntityTypeString;
```

Comment:
After this, it might make sense to rename that column in the database (for example `series_entity_type`) and similarly for the `user_collection` table (`collection_entity_type`) and review these types.
What do you think?

Reply:
Yeah, that would definitely be better in the long term.

So for now you would suggest to get rid of the `entityType` property for all parsed entity models other than `ParsedSeries` as I had planned to do?
It's probably the simplest solution for now as there are no occurrences of these types outside of the importer project.
Doing so has one minor disadvantage though, I can no longer do the following simplification:
```ts
export function getAdditionalEntityProps(
	entityData: ParsedEntity
) {
	switch (entityData.entityType) {
		case 'Author': {
			const {typeId, genderId, beginAreaId, beginDate, endDate,
				ended, endAreaId} = entityData; // "as ParsedAuthor" no longer necessary
```

Reply:
I've worked around the issue of the conflicting use of the `entityType` property by dropping it from `ParsedEntity` and instead passing around the whole `QueuedEntity` instances (which have an unambiguous  `entityType` property and also contain `data: ParsedEntity`) where necessary (1e155abdc91446b02abd0638f19d09d68689b5f9).

Later when we have better names for the database columns (of series and collections), we could consider reintroducing the `ParsedEntity.entityType` property to simplify the code again and make it easier to understand.